src/marshmallow/context.py

```diff
@@ -0,0 +1,17 @@
+"""Objects related to serializtion/deserialization context"""
+
+import contextlib
+import contextvars
+
+CONTEXT = contextvars.ContextVar("context", default=None)
+
+
+class Context(contextlib.AbstractContextManager):
+    def __init__(self, context):
+        self.context = context
+
+    def __enter__(self):
+        self.token = CONTEXT.set(self.context)
+
+    def __exit__(self, *args, **kwargs):
+        CONTEXT.reset(self.token)
```

Comment:
suggestion: maybe can unify these into a single class?

```python
class Context(contextlib.AbstractContextManager):
    current_context: contextvars.ContextVar = contextvars.ContextVar("context")

    def __init__(self, context):
        self.context = context

    def __enter__(self):
        self.token = self.current_context.set(self.context)

    def __exit__(self, *args, **kwargs):
        self.current_context.reset(self.token)

    @classmethod
    def get(cls):
        return cls.current_context.get()

Reply:
thought: it might even make sense to make `current_context` private.

```python
class Context(contextlib.AbstractContextManager):
    __current_context: contextvars.ContextVar = contextvars.ContextVar("context")